src/v/net/probes.cc

```diff
@@ -110,6 +111,26 @@ void server_probe::setup_metrics(
       });
 }
 
+void server_probe::setup_public_metrics(
+  ss::metrics::metric_groups& mgs, std::string_view proto) {
+    namespace sm = ss::metrics;
+
+    if (proto.ends_with("_rpc")) {
+      proto.remove_suffix(4);
```

Comment:
I think the suffix should be removed if it exists, but always register the metric.